src/backend/drm/compositor/mod.rs

```diff
@@ -203,6 +203,15 @@ enum ScanoutBuffer<B: Buffer> {
     Cursor(GbmBuffer),
 }
 
+impl<B: Buffer> ScanoutBuffer<B> {
+    fn acquire_point(&self) -> Option<SyncPoint> {
+        if let Self::Wayland(buffer) = self {
+            return buffer.acquire_point().cloned().map(SyncPoint::from);
```

Comment:
So if a blocker is used this point will always be signalled, right? Then we don't need to create our own signalled fence.

Reply:
That was my question yesterday.

If we assume every smithay-based compositor will use `DrmSyncpointBlocker`, we don't need to get a fence from the sync point. But should create a fence that is already signaled to use as an `IN_FENCE_FD` to make sure no implicit sync occurs.

If `DmabufBlocker` is used for any surface that isn't using explicit sync, we can also submit a signaled `IN_FENCE_FD` for implict sync surfaces. (So we can just do that for every direct scanout plane.)

(If some Smithay compositors might not want to use blockers like that, they'll need a way to submit an `IN_FENCE_FD` from the acquire point.)

Reply:
> If we assume every smithay-based compositor will use DrmSyncpointBlocker, we don't need to get a fence from the sync point. But should create a fence that is already signaled to use as an IN_FENCE_FD to make sure no implicit sync occurs.

If DmabufBlocker is used for any surface that isn't using explicit sync, we can also submit a signaled IN_FENCE_FD for implict sync surfaces. (So we can just do that for every direct scanout plane.)

So maybe we can make these blockers attach a hint on the `WlSurface`, that signals the `DrmCompositor`, that the buffer should be ready? And just fall back to implicit sync (no `IN_FENCE_FD`) otherwise?

I suppose no smithay compositor would want to use a different fence for a buffer. At least not when using the DrmCompositor, which just does direct scanout or submits an `IN_FENCE_FD` for compositing.

Anything else that might need custom fences (e.g. multi-gpu transfers directly scanned out or composing onto a plane) is not supported right now by the `DrmCompositor` anyway.

Reply:
> Though it might be simpler for DrmCompositor to take a bool of whether or not to assume buffers are already ready.

Yeah, though then we need good docs to explain when it is safe to pass that bool. (And use the fences from the acquire points otherwise?)